src/csharp/Microsoft.Spark/Utils/UdfUtils.cs

```diff
@@ -103,6 +104,15 @@ internal enum PythonEvalType
                 {typeof(DoubleArray), "double"},
                 {typeof(StringArray), "string"},
                 {typeof(BinaryArray), "binary"},
+
+                {typeof(PrimitiveDataFrameColumn<bool>), "boolean"},
+                {typeof(PrimitiveDataFrameColumn<byte>), "byte"},
+                {typeof(PrimitiveDataFrameColumn<short>), "short"},
+                {typeof(PrimitiveDataFrameColumn<int>), "integer"},
+                {typeof(PrimitiveDataFrameColumn<long>), "long"},
+                {typeof(PrimitiveDataFrameColumn<float>), "float"},
+                {typeof(PrimitiveDataFrameColumn<double>), "double"},
+                {typeof(ArrowStringDataFrameColumn), "string"},
```

Comment:
what about `binary`?

Reply:
We actually don't support binary yet in DataFrame.